### PR TITLE
Issue #1311 sprinkling from imod5 data

### LINKED ISSUE #1311: Add ``imod.msw.Sprinkling.from_imod5_data``
Based on @HendrikKok's conversion prototype:

What I didn't really get in the script yet was why the ``"artificial_recharge_capacity"`` grid is not taken there, but instead it is set to 0.001 * ``'rural_infiltration_capacity'``.

Requirements
---------------

- Derive ``max_abstraction_groundwater`` grid, where artificial_recharge == 1
- Derive ``max_abstraction_surfacewater`` grid, where artificial_recharge == 2
- If groundwater extraction, then set surface water extraction to 0 
- Convert recharge artificial layer grid to ``LayeredWell`` package
- Wells in inactive layers should deactivated.
- Throw ``NotImplementedError`` if dataframe provided instead of grid

### PR DESCRIPTION
**Pull request by JoerivanEngelen**
Fixes #1311

# Description

Adds the following:

- ``imod.mf6.LayeredWell.from_imod5_cap_data``, to construct LayeredWells for MODLOW6 from the CAP package in iMOD5 Data. Throws an error if data is provided as point data (IPF, which is read as pandas DataFrame). This has rate 0.0, actual rates will be inserted by in the coupling scheme of iMOD Coupler.
- ``imod.mf6.GroundwaterFlowModel.from_imod5_data`` adds Sprinkling well if CAP package present. This does not affect MODFLOW6 model if not coupled to MetaSWAP: The wells have rates 0.0, so will not affect the simulation results.
- ``Sprinkling.from_imod5_data`` to construct a Sprinkling package from iMOD5 data.

# Checklist

- [x] Links to correct issue
- [x] Update changelog, if changes affect users
- [x] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [x] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: imod/tests/test_msw/test_sprinkling.py [hunk 1]
```diff
@@ -269,3 +270,40 @@ def test_simple_model_1_subunit(fixed_format_parser):
...
+
+def test_sprinkling_from_imod5_data__points(cap_data_sprinkling_points):
+    with pytest.raises(NotImplementedError):
+        msw.Sprinkling.from_imod5_data(cap_data_sprinkling_points)
+
+
+def test_sprinkling_from_imod5_data__grid(cap_data_sprinkling_grid):
+    # Arrange
+    # fmt: off
+    np.nan = nan
```
Comment (Manangka) [inline imod/tests/test_msw/test_sprinkling.py:283]: Why are you ding this?

**Review by Manangka — APPROVED** [review]

File: imod/tests/test_msw/test_sprinkling.py (hunk 1 above)
Comment (JoerivanEngelen) [reply imod/tests/test_msw/test_sprinkling.py:283]: That indeed is an error, I think I wanted to do nan = np.nan, to nicely outline the matrices.

Comment (JoerivanEngelen) [reply imod/tests/test_msw/test_sprinkling.py]: Fixed

**Comment by JoerivanEngelen** [comment]
For some reason the example CI breaks online, but it works locally

**Pull request merged**